experiment/simulation/js/animation-parallel-register.js

```diff
@@ -0,0 +1,807 @@
+    textO4.textContent = textI4.textContent;
+}
+let temp = 0;
+function outputHandler() {
+    temp = textO4.textContent;
+    textO4.textContent = textO3.textContent;
+    textO3.textContent = textO2.textContent;
+    textO2.textContent = textO1.textContent;
+    textO1.textContent = temp;
+}
+function set(a) {
+    gsap.set(a, {
+
+        fill: "#eeeb22"
+    });
+}//output 0
+function unset(a) {
+    gsap.set(a, {
+
+        fill: "#29e"
+    });
+}//output 1
+function appendI1() {
+    if (textI1.textContent != 0 && tl.progress() == 0) {
+        serialDisappear1();
+        textI1.textContent = 0;
+        svg.appendChild(textI1);
+        gsap.set(textI1, {
+            x: 46,
+            y: 54
+        });
+        gsap.set(I1, {
+
+            fill: "#eeeb22"
+        });
+        free();
+        serialVisible1();
+        errno();
+        setter(textI1.textContent, serialDot1);
+
+        OBSERV.innerHTML = "I1 is set to 0";
+    }
+    else if (textI1.textContent != 1 && tl.progress() == 0) {
+        appendI1To1();
+    }
+
+
+
+
+}
+function appendI1To1() {
+    serialDisappear1();
+    textI1.textContent = 1;
+    svg.appendChild(textI1);
+    gsap.set(textI1, {
+        x: 46,
+        y: 54
+    });
+    gsap.set(I1, {
+
+        fill: "#29e"
+    });
+    free();
+    serialVisible1();
+    errno();
+    setter(textI1.textContent, serialDot1);
+
+    OBSERV.innerHTML = "I1 is set to 1";
+
+}
+function appendI2() {
+    if (textI2.textContent != 0 && tl.progress() == 0) {
+        serialDisappear2();
+        textI2.textContent = 0;
+        svg.appendChild(textI2);
+        gsap.set(textI2, {
+            x: 46,
+            y: 254
+        });
+        gsap.set(I2, {
+
+            fill: "#eeeb22"
+        });
+        free();
+        serialVisible2();
+        errno();
+        setter(textI2.textContent, serialDot2);
+
+        OBSERV.innerHTML = "I2 is set to 0";
+    }
+    else if (textI2.textContent != 1 && tl.progress() == 0) {
+        appendI2To1();
+    }
+
+
+
+
+}
+function appendI2To1() {
+    serialDisappear2();
+    textI2.textContent = 1;
+    svg.appendChild(textI2);
+    gsap.set(textI2, {
+        x: 46,
+        y: 254
+    });
+    gsap.set(I2, {
+
+        fill: "#29e"
+    });
+    free();
+    serialVisible2();
+    errno();
+    setter(textI2.textContent, serialDot2);
+
+    OBSERV.innerHTML = "I2 is set to 1";
+
+}
+function appendI3() {
+    if (textI3.textContent != 0 && tl.progress() == 0) {
+        serialDisappear3();
+        textI3.textContent = 0;
+        svg.appendChild(textI3);
+        gsap.set(textI3, {
+            x: 46,
+            y: 454
+        });
+        gsap.set(I3, {
+
+            fill: "#eeeb22"
+        });
+        setter(textI3.textContent, serialDot3);
+        free();
+        serialVisible3();
+        errno();
+
+
+
+        OBSERV.innerHTML = "I3 is set to 0";
+    }
+    else if (textI3.textContent != 1 && tl.progress() == 0) {
```

Comment:
== and != is not what we recommend for comparison. Please go through the coding guidelines